### PR TITLE
Introduce basic admission webhook checks to handle error case

### PR DESCRIPTION
**Pull request by nanzhong**
Webhooks that point to missing services or namespaces is a pretty general case not specific to doks that we should catch.

This also cleans up the error case in the doks specific check that would in result in failed clusterlint run.

### CODE REVIEW

File: checks.md [hunk 1]
```diff
@@ -207,6 +207,80 @@ spec:
...
+  - v1beta1
+  timeoutSeconds: 1
+  failurePolicy: Fail
+  namespaceSelector:
+    matchExpressions:
+      - key: "skip-webhooks"
+        operator: "DoesNotExist"
+```
+
+- Name: `admission-controller-webhook-replacement`
```
Comment (varshavaradarajan) [inline checks.md]: Add heading: `## Admission Controller Webhook Replacement`

File: checks/basic/admission_controller_webhook.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+			}
+		}
+	}
+
+	for _, config := range objects.MutatingWebhookConfigurations.Items {
+		for _, wh := range config.Webhooks {
+			if wh.ClientConfig.Service != nil {
+				// Ensure that the service (and its namespace) that is configure actually exists.
+				var svcNamespace *v1.Namespace
+				for _, ns := range objects.Namespaces.Items {
```
Comment (varshavaradarajan) [inline checks/basic/admission_controller_webhook.go]: May be we can extract this into a method? The `ClientConfig` object is the same for both `ValidatingWebhook` and `MutatingWebhook`

File: checks/basic/admission_controller_webhook.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+						Message:  fmt.Sprintf("Mutating webhook %s is configured against a service in a namespace that does not exist.", wh.Name),
+						Kind:     checks.MutatingWebhookConfiguration,
+						Object:   &config.ObjectMeta,
+						Owners:   config.ObjectMeta.GetOwnerReferences(),
+					})
+					continue
+				}
+
+				svcExists := false
+				for _, svc := range objects.Services.Items {
```
Comment (varshavaradarajan) [inline checks/basic/admission_controller_webhook.go]: Same as above: may be we can extract this into a method?

**Comment by varshavaradarajan** [comment]
lgtm

@nanzhong - I'll create a release with your change, #68 and #70 once this PR is merged.

**Comment by nanzhong** [comment]
@varshavaradarajan awesome, thanks!

**Pull request merged**